tesseract-core/src/schema.rs

```diff
@@ -91,6 +91,38 @@ pub struct Cube {
     pub measures: Vec<Measure>,
 }
 
+impl Cube {
+    /// Returns a Vec<String> of all the dimension name options for a given Cube.
+    pub fn get_all_dimension_names(&self) -> Vec<String> {
```

Comment:
It might be better to use `LevelName` from the module `names`, instead of `String`. I haven't looked at the rest of the code yet, but I would think it would make it easier to work with later, for matching/filtering on a dimension.

Or, even without guessing at future usage, it's nice to have something stronger than stringly typing, and I think that `LevelName` is appropriate here.

Reply:
(Also, I think that this fn is getting all level names, instead of dimension names)